web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/tracecontext/Traceparent.java

```diff
@@ -0,0 +1,185 @@
+package com.microsoft.applicationinsights.web.internal.correlation.tracecontext;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.http.annotation.Experimental;
+
+/**
+ * This class represents the Traceparent data structure based on
+ *
+ * @author Reily Yang
+ * @author Dhaval Doshi
+ * @link https://github.com/w3c/trace-context/blob/master/trace_context/HTTP_HEADER_FORMAT.md
+ */
+@Experimental
+public class Traceparent {
+
+    /**
+     * Version number between range [0,255] inclusive
+     */
+    @VisibleForTesting
+    final int version;
+
+    /**
+     * 16 byte trace-id that is used to uniquely identify a distributed trace
+     */
+    @VisibleForTesting
+    final String traceId;
+
+    /**
+     * It is a 8 byte ID that represents the caller span
+     */
+    @VisibleForTesting
+    final String spanId;
+
+    /**
+     * An 8-bit field that controls tracing flags such as sampling, trace level etc.
+     */
+    @VisibleForTesting
+    final int traceFlags;
+
+    private Traceparent(int version, String traceId, String spanId, int traceFlags, boolean check) {
+        if (check) {
+            validate(version, traceId, spanId, traceFlags);
+        }
+        this.version = version;
+        this.traceId = traceId;
+        this.spanId = spanId;
+        this.traceFlags = traceFlags;
+    }
+
+    /**
+     * The constructor that tries to create Traceparent Object from given version, traceId, spanID
+     * and traceFlags.
+     */
+    public Traceparent(int version, String traceId, String spanId, int traceFlags) {
+        this(version, traceId != null ? traceId : randomHex(16),
+            spanId != null ? spanId : randomHex(8),
+            traceFlags, true);
+    }
+
+    /**
+     * This constructor creates a new Traceparent object having new traceId. It should only be used
+     * if the call is the starting point of distributed trace.
+     */
+    public Traceparent() {
+        this(0, randomHex(16), randomHex(8), 0, false);
+    }
+
+    public String getTraceId() {
+        return traceId;
+    }
+
+    public int getTraceFlags() {
+        return traceFlags;
+    }
+
+    public String getSpanId() {
+        return spanId;
+    }
+
+    /**
+     * Validates the given input based on W3C specifications.
+     */
+    private static void validate(int version, String traceId, String spanId, int traceFlags)
+        throws IllegalArgumentException {
+        if (version < 0 || version > 254) {
+            throw new IllegalArgumentException("version must be within range [0, 255)");
```

Comment:
Either the condition or the message is incorrect: `[0, 255)` means between 0 inclusive and 255 exclusive.

Reply:
nvm...I read the condition wrong.